app/validators/routing/new_when_rule_validator.py

```diff
@@ -101,8 +97,26 @@ def validate_rule(self, rule):
 
         if operator_name == OPERATOR_DATE:
             return TYPE_DATE
+
+        if operator_name == OPERATOR_COUNT:
+            return TYPE_NUMBER
+
         return TYPE_BOOLEAN
 
+    def validate_argument_type(self, rule, operator_name, argument_types):
+        for argument in rule[operator_name]:
+            if isinstance(argument, dict) and any(
+                operator in argument for operator in ALL_OPERATORS
+            ):
+                argument_type = self.validate_rule(argument)
+            elif isinstance(argument, dict) and "source" in argument:
+                argument_type = resolve_value_source_json_type(
+                    argument, self.questionnaire_schema.answers_with_context
+                )
+            else:
+                argument_type = python_type_to_json_type(type(argument).__name__)
+            argument_types.append(argument_type)
```

Comment:
The main job of this is to return the arg types.
- We should rename this to something like `_get_argument_types_for_operator`, also should be private.
- We shouldn't pass in `argument_types`, it should just return it the list of arg types, we shouldn't mutate what is being passed in. So I would change the signature `def _get_argument_types_for_operator(self, arguments):`